letsencrypt/client/standalone_authenticator.py

```diff
@@ -0,0 +1,519 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+"""An authenticator that doesn't rely on any existing server program,
+but instead creates its own ephemeral TCP listener on the specified port
+in order to respond to incoming DVSNI challenges from the certificate
+authority."""
+
+import zope.interface
+import zope.component
+from letsencrypt.client import CONFIG
+from letsencrypt.client import interfaces
+from letsencrypt.client.challenge_util import DvsniChall
+from letsencrypt.client.challenge_util import dvsni_gen_cert
+import os
+import sys
+import signal
+import time
+import socket
+import struct
+import Crypto.Random
+import M2Crypto.X509
+import OpenSSL.crypto
+import OpenSSL.SSL
```

Comment:
Use imports for packages and modules only. The project convention is to import as follows:
```
imports from stdlib (os, sys, ...)

imports from dependent third-party packages (e.g. M2Crypto, OpenSSL, zope.interface, ...)

imports from the letsencrypt.client.*
```

Note that each group should be separated by an empty line, and imports in the group should be sorted lexicographically.